### PR TITLE
RD-2352

### PR DESCRIPTION
**Pull request by doriaviram**

### CODE REVIEW

File: src/globals.js [hunk 1]
```diff
@@ -1,18 +1,34 @@
...
   return { addSpan, getSpans, clearSpans };
 })();
 
+export const GlobalTimer = (() => {
+  let currentTimer = undefined;
+
+  const setGlobalTimeout = (func, duration) => {
+    clearTimer();
+    currentTimer = setTimeout(func, duration);
+    currentTimer.unref();
```
Comment (theburningmonk) [inline src/globals.js:22]: why is this needed? isn't `clearTimer` going to take care of it? is this here "just in case"?

File: src/hooks/http.js [hunk 2]
```diff
@@ -219,25 +232,34 @@ export const httpRequestWrapper = originalRequestFn =>
         const traceId = getPatchedTraceId(awsXAmznTraceId);
         options.headers['X-Amzn-Trace-Id'] = traceId;
       }
-
       const requestData = parseHttpRequestOptions(options, url);
+      const requestRandomId = getRandomId();
```
Comment (theburningmonk) [inline src/hooks/http.js:236]: I don't understand the significance of `requestRandomId` in relation to the timer

File: src/tracer.js [hunk 3]
```diff
@@ -29,6 +35,21 @@ export const ASYNC_HANDLER_REJECTED = 'async_handler_rejected';
 export const NON_ASYNC_HANDLER_ERRORED = 'non_async_errored';
 export const LEAK_MESSAGE =
   'Execution leak detected. More information is available in: https://docs.lumigo.io/docs/execution-leak-detected';
+const TIMEOUT_BUFFER_MS = 500;
```
Comment (theburningmonk) [inline src/tracer.js:38]: half a second is quite a long time, why pick such a high number?

File: src/tracer.js [hunk 4]
```diff
@@ -29,6 +35,21 @@ export const ASYNC_HANDLER_REJECTED = 'async_handler_rejected';
...
+
+const setupTimeoutTimer = () => {
+  logger.debug('Timeout timer set-up started');
+  const { context } = TracerGlobals.getHandlerInputs();
+  const { remainingTimeInMillis } = getContextInfo(context);
+  if (TIMEOUT_BUFFER_MS < remainingTimeInMillis) {
+    GlobalTimer.setGlobalTimeout(async () => {
+      logger.debug('The tracer reached the end of the timeout timer');
```
Comment (theburningmonk) [inline src/tracer.js:46]: this message is confusing, how about saying something that tells the user what's actually happening? e.g. `Invocation is about to timeout, sending trace data back to Lumigo`

Comment (doriaviram) [reply src/tracer.js:46]: 👌

File: src/tracer.js (hunk 3 above)
Comment (doriaviram) [reply src/tracer.js:38]: We going with `250ms`, maybe we will reduce this later
